packages/core/src/types/errors.ts

```diff
@@ -446,6 +447,14 @@ export interface UOpTypeError {
   arg: Value<ad.Num>["tag"];
 }
 
+export interface BadShapeParamTypeError {
+  tag: "BadShapeParamTypeError";
+  path: string;
+  value: Val<ad.Num> | ShapeVal<ad.Num>;
+  expectedType: string;
+  passthrough: boolean;
+}
```

Comment:
If I understand correctly, this error type doesn't include source location information; would it be possible to add that?

Reply:
It does not contain source location information, and I'm not sure how I can obtain them ... I don't think those are in `translation` where we get the actual values for the shape parameters.